src/Lachain.Core/Blockchain/SystemContracts/NativeTokenContract.cs

```diff
@@ -259,17 +261,36 @@ private static string PrettyParam(dynamic param)
         private void Emit(string eventSignature, params dynamic[] values)
         {
             var eventData = ContractEncoder.Encode(null, values);
+            
+            List<UInt256>? topics = null;
+            var value = new byte[32];
+            if(HardforkHeights.IsHardfork_4Active(_context.Snapshot.Blocks.GetTotalBlockHeight()))
+            {
+                // from and to address is in 2nd and 3rd topic, value transferred is in data
+                topics = new List<UInt256>();
+                if(Lrc20Interface.EventTransfer == eventSignature)
+                {
+                    byte[] from = new byte[32], to = new byte[32];
+                    Buffer.BlockCopy(eventData, 0, from, 0, 32);
+                    Buffer.BlockCopy(eventData, 32, to, 0, 32);
+                    Buffer.BlockCopy(eventData, 64, value, 0, 32);
+                    topics.Add(from.ToUInt256());
+                    topics.Add(to.ToUInt256());
+                }
+            }
             var eventObj = new EventObject(
                 new Event
                 {
                     Contract = ContractRegisterer.LatokenContract,
-                    Data = ByteString.CopyFrom(eventData),
+                    Data = (Lrc20Interface.EventTransfer == eventSignature && HardforkHeights.IsHardfork_4Active(_context.Snapshot.Blocks.GetTotalBlockHeight())) ? 
```

Comment:
It will be very hard to add more hardfork if the logics are not clearly separated. 
We could do something like this: 
if(hardfork4_active()) {
    // create eventObject
}
else {
   // create eventObject
}